test/output-consistency/mzcompose.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+from materialize.mzcompose import Composition, WorkflowArgumentParser
+from materialize.mzcompose.services import (
+    Clusterd,
+    Cockroach,
+    Kafka,
+    Localstack,
+    Materialized,
+    SchemaRegistry,
+    Testdrive,
+    Zookeeper,
+)
+from materialize.output_consistency.output_consistency import (
+    run_output_consistency_tests,
+)
+
+SERVICES = [
+    Zookeeper(),
+    Kafka(),
+    SchemaRegistry(),
+    Localstack(),
+    Cockroach(setup_materialize=True),
+    Clusterd(name="clusterd1"),
+    # We use mz_panic() in some test scenarios, so environmentd must stay up.
+    Materialized(propagate_crashes=False, external_cockroach=True),
+    Testdrive(
+        volume_workdir="../testdrive:/workdir/testdrive",
+        volumes_extra=[".:/workdir/smoke"],
+        materialize_params={"cluster": "cluster1"},
+    ),
+]
+
+
+def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
+    test_name = "test-output-consistency"
+
+    with c.test_case(test_name):
+        c.workflow(test_name)
+
+
+def workflow_test_output_consistency(c: Composition) -> None:
+    """
+    Test the output consistency of different query evaluation strategies (e.g., dataflow rendering and constant folding).
+    """
+
+    c.down(destroy_volumes=True)
+    with c.override(
+        Testdrive(no_reset=True),
+    ):
+        c.up("testdrive", persistent=True)
+        c.up("materialized")
+
+    run_output_consistency_tests(c)
```

Comment:
Also would be nice to have an additional mode of running this from command line and passing a postgresql connection string, without `mzcompose` (or maybe that's just me who prefers to run things like that?)

Reply:
We need both and they are not mutually exclusive.

- writing things mzcompose-first means that it will be easier for them to go beyond the "this is a one-off script" and rapidly move to be included in the CI -- after all, once there is a mzcompose, it is only a very small step to add it to the Nightly CI. So, in this sense, using mzcompose has a disciplining , "ship-it" type of effect.
- at the same time, you can totally have an `--materialize-url` option to the mzcompose workflow. If set, it should not do the `c.up` and move directly to creating a `cursor` object out of the passed URL 

This comment of mine should be worth at least 20 eur. This is how much I paid for Lufthansa in-flight internet and it is so slow that I can not really work on the rest of the PR right now.